build-on-push/platform_ci/platform_ci/jenkins_jobs.py

```diff
@@ -0,0 +1,191 @@
+                               "github-user": self.platform_ci_repo}}
+
+        return yaml.dump([template, project], default_flow_style=False)
+
+
+class JobCommitDispatcher(object):
+    """Represents the Build-on-Push dispatcher job.
+
+    The dispatcher job monitors all changes in a DistGit repository for a given
+    component, and based on the branch where the changes were made, decides if
+    the testing scratch-build should be automatically attempted. There should be
+    a single dispatcher job per component.
+    """
+    def __init__(self, component, slave=None, platform_ci_repo=None, platform_ci_branch=None):
+        self.component = component
+        self.slave = slave
+        self.platform_ci_branch = platform_ci_branch
+        self.platform_ci_repo = platform_ci_repo
+
+    @staticmethod
+    def create_description(branch, built_targets, jenkins_url, component):
+        """Creates a build description for the dispatcher job.
+
+        Args:
+            branch: A name of the branch with the commits triggering the dispatcher
+            built_targets: A list of targets that were decided to build
+            jenkins_url: The URL of the Jenkins master where the description will be viewed.
+            component: A name of the component
+        Returns:
+            A string with the HTML fragment suitable to be used as a Jenkins job build
+                description
+        """
+        lines = ["<strong>Dist-git branch</strong>: {0.name} ({0.type} branch)".format(branch)]
+
+        if not built_targets:
+            lines.append("No brew build was issued ({0.name} is not handled by CI)".format(branch))
+        else:
+            trigger_job_template = '<strong>Triggered job: </strong><a href="{0}/job/{1}">Worker job for branch {2}</a>'
+            worker_job_name = JobBuildOnCommit.create_job_name(component, branch.name)
+            trigger_job_message = trigger_job_template.format(jenkins_url, worker_job_name, branch.name)
+            lines.append(trigger_job_message)
+
+        return "<p>{0}</p>".format("<br>".join(lines))
+
+    @staticmethod
+    def create_job_name(component):
+        return "ci-{0}-dispatcher-commit".format(component)
+
+    @property
+    def name(self):
+        """Returns the name of the job represented by an instance."""
+        return JobCommitDispatcher.create_job_name(self.component)
+
+    @property
+    def display_name(self):
+        """Returns the (human-oriented) label of a job represented by an instance."""
+        return "{0}: Schedule Brew build".format(self.component)
+
+    def as_yaml(self):
+        """Returns a YAML string usable for instantiation of the JJB template.
+
+        The output is the YAML with values which can be used to instantiate
+        the 'ci-dispatcher-commit' template to create a Jenkins job using the
+        Jenkins Job Builder.
+        """
+
+        # TODO: These settings should be centralized somewhere else
```

Comment:
I'll need to think better about this suggestion. At the very least, the TODO comment should be more informative about what is needed and why it was not done in the first place. Perhaps file a GH issue in that case.